### PR TITLE
Tutorial: Live queries

### PR DESCRIPTION
**Pull request by sdorkenw**
- Extended synapse tutorial to include live queries and new endpoints
- Added new endpoint to query root timestamp
- Added logic to retrieve "latest" and "original" root ids
- Revised query to check for expired _and_ too recent status of root ids

I am aware that the latest features are only available for PCGv1. We need to move the latest feature block to PCGv2.

### CODE REVIEW

**Comment by sdorkenw** [comment]
The test is failing due inconsistent handling of timestamps (introduced [here](https://github.com/seung-lab/AnnotationFrameworkClient/pull/36/files#diff-2aedcefafced9d0e25e4368908be9f49996ae39e3be8b63ce6d82ef406350560R648)). Okay to map all timestamps to UTC?

File: annotationframeworkclient/chunkedgraph.py [hunk 1]
```diff
@@ -467,10 +473,70 @@ def get_lineage_graph(self, root_id, timestamp_past=None, timestamp_future=None)
 
         url = self._endpoints["handle_lineage_graph"].format_map(endpoint_mapping)
         r = handle_response(self.session.get(url, params=data))
-        return r
+
+        if as_nx_graph:
```
Comment (fcollman) [inline annotationframeworkclient/chunkedgraph.py:477]: i don't think this one line difference is worth adding the dependency to this package personally.

Comment (fcollman) [reply annotationframeworkclient/chunkedgraph.py:477]: I see you are gonna use it to make client side connivence functions, i commented before reading the whole thing.

File: annotationframeworkclient/materializationengine.py [hunk 2]
```diff
@@ -644,46 +645,62 @@ def map_filters(self, filters, timestamp, timestamp_past):
...
+        if len(root_ids) == 0:
             return [None, None, None], {}
-        vals = np.unique(np.concatenate(vals))
-        filter_vals_latest = self.cg_client.is_latest_roots(vals, timestamp=timestamp)
-        if not np.all(filter_vals_latest):
-            not_latest = vals[~filter_vals_latest]
+        root_ids = np.unique(np.concatenate(root_ids))
+
+        filter_timed_end = self.cg_client.is_latest_roots(root_ids, timestamp=timestamp)
+        filter_timed_start = self.cg_client.get_root_timestamps(root_ids) < timestamp
```
Comment (fcollman) [inline annotationframeworkclient/materializationengine.py:667]: if this isn't implemented in Minnie we can't merge and release it to pypi till it is.

File: FlyWireSynapseTutorial.ipynb [hunk 3]
```diff
@@ -1002,18 +1049,521 @@
...
+   ],
+   "source": [
+    "%%time \n",
     "\n",
-    "Neuroglancer shows the most recent version of the segmentation by default. Neurons that have been updated since a materialized version are not included in a table of that version. To reconcile this, users need to look up root ids for their data with a timestamp. \n",
+    "# Code to retrieve a root id that will work with this query. See the next section for more details\n",
+    "latest_roots = client.chunkedgraph.get_latest_roots(720575940627185911, timestamp_future=timestamp_now)\n",
+    "latest_roots\n",
     "\n",
-    "We generally recommend storing annotations as points in space as these can be mapped to root ids easily (that's basically what materialization is). Soon, users will be able to create their own annotation tables and CAVE will provide fitting root ids automatically. Still, use cases will arrive that require a mnual materialization by the user:\n",
```
Comment (fcollman) [inline FlyWireSynapseTutorial.ipynb:1009]: typo here

Comment (fcollman) [reply FlyWireSynapseTutorial.ipynb:1009]: mnual

**Comment by fcollman** [comment]
Yes I think its ok to map all timestamps to UTC, but I don't see how the tests didn't work out... I think you need to add some mocks to the tests in order to for the new endpoint to be tested properly.

**Review by fcollman — APPROVED** [review]

File: annotationframeworkclient/materializationengine.py (hunk 2 above)
Comment (sdorkenw) [reply annotationframeworkclient/materializationengine.py:672]: just added that to minnie and deployed

**Pull request merged**